--- a/track_local_static.go	2026-09-11 13:31:34
+++ b/track_local_static.go	2026-09-11 13:31:34
@@ -9,11 +9,14 @@
 	"strings"
 	"sync"
 
+	"github.com/google/uuid"
 	"github.com/kulikov0/headless-client/webrtc/internal/util"
 	"github.com/kulikov0/headless-client/webrtc/pkg/media"
 	"github.com/pion/rtp"
 )
 
+const msidNoStreamID = "-"
+
 // trackBinding is a single bind for a Track
 // Bind can be called multiple times, this stores the
 // result for a single bind call so that it can be used when writing.
@@ -45,8 +48,8 @@
 	t := &TrackLocalStaticRTP{
 		codec:    c,
 		bindings: []trackBinding{},
-		id:       id,
-		streamID: streamID,
+		id:       uuid.NewString(),
+		streamID: msidNoStreamID,
 	}
 
 	for _, option := range options {
